addons/web_editor/controllers/main.py

```diff
@@ -696,3 +723,11 @@ def bus_broadcast(self, model_name, field_name, res_id, bus_data):
         channel = (request.db, 'editor_collaboration', model_name, field_name, int(res_id))
         bus_data.update({'model_name': model_name, 'field_name': field_name, 'res_id': res_id})
         request.env['bus.bus']._sendone(channel, 'editor_collaboration', bus_data)
+
+    @http.route("/web_editor/ensure_common_history", type="json", auth="user")
+    def ensure_common_history(self, model_name, field_name, res_id, history_ids):
+        record = request.env[model_name].browse([res_id])
+        try:
+            prevent_diverging_history(record, field_name, history_ids)
```

Comment:
Is there added value in having this route taking different parameters than the write one ? Because of that, we have two functions that prevent diverging history: `prevent_diverging_history` and `prevent_diverging_history_on_write`. Why not have this controller here take the html of the editable like `write` ? That way we could put the code of `prevent_diverging_history` into `prevent_diverging_history_on_write`, rename it into `prevent_diverging_history` and then have only one function that handles the diverging history stuff.